modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoClientCodegen.java

```diff
@@ -448,26 +448,44 @@ public CodegenProperty fromProperty(String name, Schema p) {
             for (CodegenParameter p : op.allParams) {
                 p.vendorExtensions.put("x-go-example", constructExampleCode(p, modelMaps, processedModelMaps));
             }
+            processedModelMaps.clear();
         }
 
-        processedModelMaps.clear();
         for (CodegenOperation operation : operationList) {
+            boolean needTimeImport = false;
             for (CodegenParameter cp : operation.allParams) {
                 cp.vendorExtensions.put("x-go-example", constructExampleCode(cp, modelMaps, processedModelMaps));
+                if (cp.isDateTime || cp.isDate) { // datetime or date
+                    needTimeImport = true;
+                }
+            }
+            if (needTimeImport) {
+                operation.vendorExtensions.put("x-go-import", "    \"time\"");
             }
+            processedModelMaps.clear();
         }
 
         return objs;
     }
 
     private String constructExampleCode(CodegenParameter codegenParameter, HashMap<String, CodegenModel> modelMaps, HashMap<String, Integer> processedModelMap) {
         if (codegenParameter.isArray) { // array
-            return codegenParameter.dataType + "{" + constructExampleCode(codegenParameter.items, modelMaps, processedModelMap) + "}";
+            String prefix = codegenParameter.dataType;
+            String dataType = codegenParameter.dataType.substring(2);
+            if (modelMaps.containsKey(dataType)) {
+                prefix = "[]" + goImportAlias + "." + dataType;
+            }
+            return prefix + "{" + constructExampleCode(codegenParameter.items, modelMaps, processedModelMap) + "}";
         } else if (codegenParameter.isMap) {
-            return codegenParameter.dataType + "{ \"key\": " + constructExampleCode(codegenParameter.items, modelMaps, processedModelMap) + "}";
+            String prefix = codegenParameter.dataType;
+            String dataType = codegenParameter.dataType.substring(13); // map[string][]
```

Comment:
I'd suggest using `"map[string][]".length()` instead of `13`. It's self-commenting and also prevents off-by-one errors. (There are couple instances of this, so please fix them all)

Reply:
Good thinking. I tried `removeStart ` which seems even more clear, let me know if it works.